File: Demos/OC/UFileSDKDemo/UFileSDKDemo/file/FileHeadFileVC.m
//
//  FileHeadFileVC.m
//  UFileSDKDemo
//
//  Created by ethan on 2018/12/5.
//  Copyright © 2018 ucloud. All rights reserved.
//

#import "FileHeadFileVC.h"

@interface FileHeadFileVC ()
@property (weak, nonatomic) IBOutlet UITextField *fileNameTF;
@property (weak, nonatomic) IBOutlet UITextView *resTV;

@end

@implementation FileHeadFileVC

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)processHeadFileHandler:(UFError *)ufError response:(UFHeadFile *)ufHeadFile
{
    dispatch_async(dispatch_get_main_queue(), ^{
        if (!ufError) {
            self.resTV.text = [NSString stringWithFormat:@"获取文件的HeadFile成功，服务器返回信息--> %@",ufHeadFile.description];
            return;
        }
        
        if (ufError.type == UFErrorType_Sys) {
            self.resTV.text = [NSString stringWithFormat:@"获取文件的HeadFile失败，系统错误信息--> %@",ufError.error.description];
            return;
        }
        self.resTV.text = [NSString stringWithFormat:@"获取文件的HeadFile失败，服务器返回错误信息--> %@",ufError.fileClientError.description];
    });
}

- (IBAction)onPressedButtonFileHeadFile:(id)sender {
    [self hideKeyboard];
    self.resTV.text = NULL;
    __weak typeof(self) weakself = self;
    
    [_fileClient headFileWithKeyName:self.fileNameTF.text success:^(UFError * _Nullable ufError, UFHeadFile * _Nullable ufHeadFile) {
        [weakself processHeadFileHandler:ufError response:ufHeadFile];
    }];
    
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    [self hideKeyboard];
}

- (void)hideKeyboard
{
    if ([self.fileNameTF isFirstResponder]) {
        [self.fileNameTF resignFirstResponder];
    }
    
}


@end

File: Demos/Swift/UFileSDKDemo-swift/UFileSDKDemo-swift/file/FileHeadFileVC.swift
//
//  FileHeadFileVC.swift
//  UFileSDKDemo-swift
//
//  Created by ethan on 2018/12/12.
//  Copyright © 2018 ucloud. All rights reserved.
//

import UIKit

class FileHeadFileVC: UIViewController {

    @IBOutlet weak var fileNameTF:UITextField!
    @IBOutlet weak var resTV:UITextView!
    
    var fileClient:UFFileClient = UFFileClient()
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    func processHeadFileHandler(ufError:UFError?, ufHeadFile:UFHeadFile?) -> Void
    {
        DispatchQueue.main.async {
            if ufError == nil {
                self.resTV.text = "获取文件的HeadFile成功，服务器返回信息-->" + ufHeadFile!.description
                return
            }
            
            if ufError?.type == UFErrorType_Sys {
                let error  = ufError?.error as! NSError
                self.resTV.text = "获取文件的HeadFile失败，系统错误信息-->" + error.description
                return
            }
            
            self.resTV.text = "获取文件的HeadFile失败，服务器返回错误信息-->" + (ufError?.fileClientError.description)!
            
        }
    }
    
    @IBAction func onPressedButtonFileHeadFile(_ sender: Any)
    {
        self.hideKeyboard()
        self.resTV.text = nil
        weak var weakself:FileHeadFileVC! = self
        self.fileClient.headFile(withKeyName: self.fileNameTF.text!) { (ufError:UFError?, ufHeadFile:UFHeadFile?) in
            weakself.processHeadFileHandler(ufError: ufError, ufHeadFile: ufHeadFile)
        }
    
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        self.hideKeyboard()
    }

    func hideKeyboard() -> Void
    {
        if self.fileNameTF.isFirstResponder {
            self.fileNameTF.resignFirstResponder()
        }
    }

}
